backend/index.py

```diff
@@ -79,7 +81,40 @@ def clean_expired_tokens():
 app.register_blueprint(tag_bp, url_prefix='/tag')
 
 
+socketio = SocketIO(app, cors_allowed_origins="*")
+
+users = {}
+
+@socketio.on('join_room')
+def handle_join_room(data):
+    authorized_username = data.get("authorizedUsername")
```

Comment:
It's generally not a good idea to let the client control the authorization state. Even though it might not be causing issues right now, adding some authentication checks could help prevent potential IDOR vulnerabilities.